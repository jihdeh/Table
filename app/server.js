const express = require('express');

const app = express();

app.set('port', (process.env.PORT || 3001));

app.use(express.static('client/build'));

const server = app.listen(app.get('port'));

module.exports = {
  server: server
};
