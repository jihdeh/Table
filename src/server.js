const express = require('express');

const app = express();

app.set('port', (process.env.PORT || 3001));

if (process.env.NODE_ENV === 'production') {
  app.use(express.static('../client/build'));
}

app.get('/', ((req, res)=> res.json({ hi: 'hi' })))

const server = app.listen(app.get('port'), () => {
  console.log(`Find the server at: http://localhost:${app.get('port')}/`);
});

export default server;
