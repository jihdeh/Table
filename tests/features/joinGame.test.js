import { server } from '../../app/server';
import createBrowser from '../browser/phantom';
import res from '../../client/src/resources/pages/startScreen';

let browser;

beforeAll(async () =>{
  browser = await createBrowser();
});

it('join game page matches snapshot', async () => {
  await browser.visit('/');
  const content = await browser.pageContent();
  expect(content).toMatchSnapshot();
});

it('shows "join as player" button', async () => {
  await browser.visit('/');
  const playerJoinButton = await browser.find('.player-join-button');
  expect(playerJoinButton.innerText).toBe(res.playerJoinButton);
});

it('clicking "join as player" button shows a deck of cards', async () => {
  await browser.visit('/');
  const playerJoinButton = await browser.click('.player-join-button');
  const cardDeck = await browser.find('.deck');
  expect(cardDeck.className).toBe('deck');
});

afterAll(async () => {
  await browser.exit();
  server.close();
});
