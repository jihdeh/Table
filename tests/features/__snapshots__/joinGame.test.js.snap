exports[`test join game page matches snapshot 1`] = `"<html><head></head><body><pre style=\"word-wrap: break-word; white-space: pre-wrap;\">{\"hi\":\"hi\"}</pre></body></html>"`;
